### DISCUSSION TITLE
Create3 IR sensor positioning

### DISCUSSION THREAD

**Discussion by vivianweli** [iRobotEducation/create3_docs#342] [Questions] [answered]
Hi, 

I'm wondering how the robot's 7 sensors are placed. What is the angle between the sensors? Do the front center left sensor and front right sensor share the center (90 degrees line in the unit circle)? Is the side left sensor placed at 180 degrees (unit circle)? And I notice there is no side right sensor, does that mean there is no sensor at the 0 degree line (as a unit circle)? Thank you.

**Comment by shamlian** [comment, marked as answer]
I really thought we had a nice picture of this on the docs site, but I looked, and it turns out we don't, so I'll have to get on that...

That said, the answer to your question can be found in [this URDF (Unified Robot Description Format) file](https://github.com/iRobotEducation/create3_sim/blob/b7c69013d0db241df64199cae9491286635d1bcc/irobot_create_common/irobot_create_description/urdf/create3.urdf.xacro#L170-L207) in the sim repository.

Though it's pretty hard to read, all the sensors are defined there with x, y, z offsets from the robot's center, and roll, pitch, and yaw in radians (though only the yaw is set).

All of that said: pretty picture coming soon. (And thanks for pointing out this oversight on the docs site!)

**Reply by shamlian** [reply]
Please watch https://github.com/iRobotEducation/create3_docs/issues/343 -- we'll close that once the image is posted.

**Reply by justinIRBT** [reply]
You can also see the sensor layout on Rviz2 by looking at the TF tree.  That should also show the values for the offsets in the panel.

**Comment by raphaelcherney** [comment]
PR #346 should add this to the Docs site, but adding here for easy reference. Here are the positions of the reflective IR sensors:
![ir_sensors_drawing](https://user-images.githubusercontent.com/1734916/223540312-12109be5-30e3-46b2-b893-83710d112592.svg)
